tests/integration/awslambda/test_lambda_integration.py

```diff
@@ -422,6 +597,225 @@ def verify_failure_received():
+            # depending on the parametrize values the filter (and the items to put) the lambda might be called multiple times
+            if calls > 1:
+
+                def assert_events_called_multiple():
+                    events = get_lambda_log_events(function_name, logs_client=logs_client)
+                    assert len(events) == calls
+
+                # lambda was called a second time, so new records should be found
+                retry(assert_events_called_multiple, retries=max_retries)
+            else:
+                # lambda wasn't called a second time, so no new records should be found
+                retry(assert_lambda_called, retries=max_retries)
+
+        finally:
+            if event_source_uuid:
+                lambda_client.delete_event_source_mapping(UUID=event_source_uuid)
+
+    @pytest.mark.aws_validated
+    @pytest.mark.parametrize(
+        "filter",
+        [
+            "single-string",
+            '[{"eventName": ["INSERT"=123}]',
+        ],
+    )
+    def test_dynamodb_invalid_event_filter(
+        self,
+        create_lambda_function,
+        lambda_client,
+        dynamodb_client,
+        dynamodb_create_table,
+        lambda_su_role,
+        wait_for_dynamodb_stream_ready,
+        filter,
+    ):
+
+        function_name = f"lambda_func-{short_uid()}"
+        table_name = f"test-table-{short_uid()}"
+        event_source_uuid = None
+
+        try:
+
+            create_lambda_function(
+                handler_file=TEST_LAMBDA_PYTHON_ECHO,
+                func_name=function_name,
+                runtime=LAMBDA_RUNTIME_PYTHON37,
+                role=lambda_su_role,
+            )
+            dynamodb_create_table(table_name=table_name, partition_key="id")
+            _await_dynamodb_table_active(dynamodb_client, table_name)
+            stream_arn = dynamodb_client.update_table(
+                TableName=table_name,
+                StreamSpecification={"StreamEnabled": True, "StreamViewType": "NEW_AND_OLD_IMAGES"},
+            )["TableDescription"]["LatestStreamArn"]
+            wait_for_dynamodb_stream_ready(stream_arn)
+            event_source_mapping_kwargs = {
+                "FunctionName": function_name,
+                "BatchSize": 1,
+                "StartingPosition": "TRIM_HORIZON",
+                "EventSourceArn": stream_arn,
+                "MaximumBatchingWindowInSeconds": 1,
+                "MaximumRetryAttempts": 1,
+                "FilterCriteria": {
+                    "Filters": [
+                        {"Pattern": filter},
+                    ]
+                },
+            }
+
+            with pytest.raises(Exception) as expected:
+                lambda_client.create_event_source_mapping(**event_source_mapping_kwargs)
+            expected.match(INVALID_PARAMETER_VALUE_EXCEPTION)
+
+        finally:
+            if event_source_uuid:
+                lambda_client.delete_event_source_mapping(UUID=event_source_uuid)
+
```

Comment:
NIT: We never assign `event_source_uuid` in this test. Maybe you can add the assignment for `create_event_source_mapping` which would be the safe-version in case the expected exception is not thrown.